testng-core/src/test/java/org/testng/xml/internal/TestNamesMatcherTest.java

```diff
@@ -54,7 +54,7 @@ public void testCloneIfContainsTestsWithNamesMatchingAnyChildSuites(
       dataProvider = "getData")
   public void testCloneIfContainsTestsWithNamesMatchingAnyNegativeCondition(
       XmlSuite xmlSuite, List<String> names) {
-    TestNamesMatcher testNamesHelper = new TestNamesMatcher(xmlSuite, names);
+    new TestNamesMatcher(xmlSuite, names);
```

Comment:
Useless line

Reply:
No it's not. The constructor of `TestNamesMatcher` internally does some edit checks and throws an exception and this test is vetting out that the exception is actually getting thrown or not.

Reply:
But that is a side effect and breaks oop.
Could you considere to move the logic into a method used here and into the constructor.

Reply:
I didnt understand you here @juherr. The [current implementation](https://github.com/testng-team/testng/blob/master/testng-core/src/main/java/org/testng/xml/internal/TestNamesMatcher.java#L31-L35) invokes a method named [cloneIfContainsTestsWithNamesMatchingAny](https://github.com/testng-team/testng/blob/789171a1aadc3e3e5ba74b1fa4dcbc34a544012e/testng-core/src/main/java/org/testng/xml/internal/TestNamesMatcher.java#L44C16-L56) 

This call is being done via constructor invocation. Are you suggesting that we move this call to some sort of a lazy invocation and leave the constructor to just member initialisation ? I believe that this rule is currently violated in multiple places within TestNG.

Reply:
The object construction is useless here because we don't use any object.
The only reason to use it is because we know there are some side effects.
It will be more logical to call a real function instead.

It could be fixed every where possible

Reply:
> It will be more logical to call a real function instead.

This wont be possible, because then it would change the semantics of what this class represents. We will have to alter all calls to explicitly call this new validation method. I believe that the validation is part of this class internals and the outside caller doesn't need to know anything about it.

Validating input parameters as part of Object instantiation, AFAIK is NOT a huge problem. It's a general practical to add semantic validations for input parameters passed into the constructor. This method `cloneIfContainsTestsWithNamesMatchingAny` is also doing such a semantic validation.

So can you please help me understand what is the real concern here so it can be addressed?

Reply:
@juherr - Can you please let me know if this concern is addressed and if this PR can be proceeded further with approval and merge?